### ISSUE TITLE
Feature Request: Support for multiple databases with the Citus integration

### ISSUE THREAD

**Issue by scottybrisbane** [zalando/patroni#2567]
**Describe the feature request**

We’re actively using Citus + Patroni with multiple "citus-enabled" databases in production and would really like to be able to use the new [Citus integration with Patroni](https://patroni.readthedocs.io/en/latest/citus.html) in our setup. Currently this integration only supports clusters with a single "citus-enabled" database which prevents us from using the new functionality.

**Comment by lkral-navmatix** [comment]
We would be very interested in using multiple Citus enabled databases with Patroni as well.
I am not currently familiar with the codebase but if there is something we could do to help move this forward, let me know.

**Comment by DBexpert** [comment]
I'm also very interested in this feature (and in our case it only need to support few databases - 5 or so; no more than 10)

**Comment by CyberDem0n** [comment]
> few databases - 5 or so; no more than 10

Usually people create many databases in a single cluster because they are small or because the host where cluster runs has a lot of hardware resources. In both cases it is about efficiently utilizing resources.

Citus solves a different problem. You have a lot of data so it doesn't fit into a single machine and you want to spread this data across many of machines.

Therefore, creating multiple databases (especially so many) with a Citus extension is an antipattern. Either you have to few data and you don't need Citus, or you better run multiple Citus clusters, with one DB per cluster.

In other words, this feature will never be a top priority.

**Comment by semirhadzic** [comment]
> > few databases - 5 or so; no more than 10
> 
> Usually people create many databases in a single cluster because they are small or because the host where cluster runs has a lot of hardware resources. In both cases it is about efficiently utilizing resources.
> 
> Citus solves a different problem. You have a lot of data so it doesn't fit into a single machine and you want to spread this data across many of machines.
> 
> Therefore, creating multiple databases (especially so many) with a Citus extension is an antipattern. Either you have to few data and you don't need Citus, or you better run multiple Citus clusters, with one DB per cluster.
> 
> In other words, this feature will never be a top priority.

Hi @CyberDem0n,

I believe your answer is antipattern per se.

Citus support multi database, we only kindly ask that  Patroni supports what Citus already support.
There is more benefits to Citus + Patroni than only spread of database storage, most of all parallelized performance for queries spread on multi hosts for mid-large databases.
Especially if you use read only "Patroni" replicas for that.

I would not go into host / cluster / storage /database strategies (for which you are  correct)  but there is real need for this Patroni feature.

Would it be possible to raise up priority for this request?

**Comment by CyberDem0n** [comment]
> Would it be possible to raise up priority for this request?

I am sorry, but we don't have plans to work on it.
Patroni is an open-source project, if you really need this feature feel free to contribute it and we will happily review your work.
